### PR TITLE
Feature/connect forms

### LINKED ISSUE #5: As a student, when I click on a book icon
- [ ] I can log minutes ('bookmark')
- [ ] I can log pages
- [ ] I can make a comment 
- [ ] There is a progress bar telling me how far I've made it through the book
- [ ] There is a button to finish the book or abandon the book

### LINKED ISSUE #7: As a student, when I click 'finish' on a book, there is a form for me to rate the book on a scale of 1-5 and leave a review

### PR DESCRIPTION
**Pull request by kellydinneen**
### What does this PR do? 
- opens ` BookDetails` page when user clicks book icon
- connects `FinishBookForm` and `NewBookMarkForm` to `BookDetails` page
- drafts apiCalls for POST-ing bookmarks, PATCH-ing finished books, and 

### Is this a feature or Fix/Refactor?  
- Feature

### CODE REVIEW

File: src/NewBookMarkForm/NewBookMarkForm.js [hunk 1]
```diff
@@ -1,56 +1,16 @@
 import React, { useState } from 'react';
+import { useForm } from "react-hook-form";
```
Comment (thaomonster) [inline src/NewBookMarkForm/NewBookMarkForm.js:2]: This is great! Thanks for updating it to use the new form hooks.

File: src/BookDetails/BookDetails.js [hunk 2]
```diff
@@ -1,21 +1,70 @@
...
+            <article className='book-options'>
+              {!displayFinishBookForm && <button onClick={() => setDisplayFinishBookForm(true)}>Finish Book</button>}
+              {displayFinishBookForm &&
+                <>
+                  <FinishBookForm book={book}/>
+                  <button onClick={() => setDisplayFinishBookForm(false)}>Cancel</button>
+                </>
+              }
+              <button>Abandon Book</button>
+            </article>
```
Comment (conconartist) [inline src/BookDetails/BookDetails.js:64]: Awesome.  The form looks great! Can't wait to get everything hooked up.

File: src/CurrentBookRainbow/CurrentBookRainbow.js [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+
+        const currentReads = data.data.slice(0, 5);
         const rainbowBox = d3.select(".rainbowBox")
         const bookPositionScale = d3.scaleQuantize()
             .domain([0, 1])
             .range([[50, 550], [90, 500], [120, 430], [130, 360], [160, 300], [200, 240], [235, 160], [280, 100], [325, 30], [380, -35]])
         const xScale = d3.scaleLinear()
             .range([70, 1000])
-        const tickLabels = data.data.map(d => {
+        const tickLabels = currentReads.map(d => {
```
Comment (conconartist) [inline src/CurrentBookRainbow/CurrentBookRainbow.js:20]: Nice refactor here.

File: src/CurrentBookRainbow/CurrentBookRainbow.js [hunk 4]
```diff
@@ -24,40 +28,53 @@ const CurrentBookRainbow = ({ data }) => {
...
+            .attr("height", "75")
+            .attr("width", "75")
+
+        const bookIcons = rainbowBox.selectAll(".bookImage")
+            .data(currentReads)
+            .enter().append("image")
+            .attr("class", "bookImage")
+            .attr("xlink:href", bookImg)
+            .attr("x", (d, i) => 50 + 175 * i)
+            .attr("y", '680')
```
Comment (conconartist) [inline src/CurrentBookRainbow/CurrentBookRainbow.js:55]: Glad you were able to figure out how to get the book images to render properly!

File: src/NewBookMarkForm/NewBookMarkForm.js [hunk 5]
```diff
@@ -2,15 +2,35 @@ import React, { useState } from 'react';
...
+  };
+  console.log(errors);
+
+  return (
+   <form onSubmit={handleSubmit(onSubmit)}>
+     <input type="number" placeholder="What page did you finish on?" {...register("What page did you finish on?", {required: true})} />
+     <input type="number" placeholder="How many minutes did you read for?" {...register("How many minutes did you read for?", {required: true})} />
+     <input type="text" placeholder="Write down any thoughts or notes you have:" {...register} />
+     <select {...register}>
+        <option value="😮">😮</option>
```
Comment (thaomonster) [inline src/NewBookMarkForm/NewBookMarkForm.js:42]: Love these emojis that a student can choose to show how they are feeling

File: src/FinishBookForm/FinishBookForm.js [hunk 6]
```diff
@@ -1,34 +1,41 @@
...
+          <option value="3"> ⭐️⭐️⭐️</option>
+          <option value="4"> ⭐️⭐️⭐️⭐️</option>
+          <option value="5"> ⭐️⭐️⭐️⭐️⭐️</option>
+        </select>
+        <label className='finish-form-label'>Tell your friends a little bit about it:</label>
+        <input type="text" placeholder="review" {...register("review", {required: true, maxLength: 500})} />
+
+        <input type="submit" />
+      </form>
+    );
```
Comment (conconartist) [inline src/FinishBookForm/FinishBookForm.js:40]: Love this functionality.  Great work on this.

File: src/NewBookForm/NewBookForm.js [hunk 7]
```diff
@@ -1,61 +1,35 @@
...
-          </button>
+        <section className='add-a-book-form'>
+          <button onClick={() => setDisplay(false)}>Cancel</button>
+          <form className='search-form' onSubmit={handleSubmit(onSubmit)}>
+            <label>What's the title of your book?</label>
+            <input type="text" placeholder="title" {...register("title", {required: true, maxLength: 100})} />
+            <label>Who is the author of your book?</label>
+            <input type="text" placeholder="author" {...register("author", {required: true, maxLength: 100})} />
+            <input type="submit" />
+          </form>
```
Comment (conconartist) [inline src/NewBookForm/NewBookForm.js:29]: The refactoring looks super clean and easy to read. Good job!  The layout makes sense-- looking forward to getting this styled.

File: src/NewBookMarkForm/NewBookMarkForm.js [hunk 8]
```diff
@@ -1,56 +1,54 @@
...
+        <option value=" 🧐"> 🧐</option>
+        <option value=" 😆"> 😆</option>
+        <option value=" 😢"> 😢</option>
+        <option value=" 😡"> 😡</option>
+        <option value=" 😨"> 😨</option>
+        <option value=" 😊"> 😊</option>
+        <option value=" 🥰"> 🥰</option>
+      </select>
```
Comment (conconartist) [inline src/NewBookMarkForm/NewBookMarkForm.js:50]: Awesome! I'm so glad we can add emoji responses now!

**Review by conconartist — APPROVED** [review]

**Review by thaomonster — APPROVED** [review]
Everything looks good! Thanks for doing all this.

**Pull request merged**